### DISCUSSION TITLE
CHANGELOG.md entries and conflicts

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#775] [Developers]
With recent events in mind, how are we going to do with the entries in the CHANGELOG? There are lots of small bugs that needs fixing but many of them are fairly insignificant in the larger picture. Do we even need to document every small typo correction in the CHANGELOG or is the commit history enough for one line/character corrections?

**Comment by oleg68** [comment]
> With recent events in mind, how are we going to do with the entries in the CHANGELOG?

There are two different questions:

1. Which changes to include or not to include to CHANGELOG?
2. How to work with CHANGELOG to avoid or minimize conflicts from technical point of view?

> Do we even need to document every small typo correction in the CHANGELOG or is the commit history enough for one line/character corrections?

I think, no. We add the changes to CHANGELOD what we want appear in the [Release Notes](https://github.com/GrandOrgue/grandorgue/releases) and be visible for end users.

Cases when a change shoudn't appear in the CHANGELOG
1. Internal code refactoring without changing an external behaviour (for example, #772)
2. Fixing a bug if the bug has been introduced in a previous commit in the same release (there wasn't releases between introducing the bug and fixing). The same if a feature has not be done with previous commits

Cases when a change must appear in the CHANGELOG:
1. When the visible behaviour of GO has been changed
2. When the change resolves an existing issue on GitHub

Intermediate cases when a change may or may not be added
1. Changes related only to the development or the build process (for example, #767, #773)
2. Fixing typo in the documentation or in the translation

Earlier, when I was alone in commiting GO, I preferred to register changes in all doubts. But now, when there are several commiters, it may cause conflicts, so we may change this practice.

**Reply by larspalo** [reply]
> Cases when a change must appear in the CHANGELOG:
> 
>     1. When the visible behaviour of GO has been changed
> 
>     2. When the change resolves an existing issue on GitHub

I agree with this. Visible changes should of course be documented, but changes that are of underlying, technical art, that never will be seen by the users (but for instance only by developers not seeing any warnings or errors at compile time anymore) might as well be left out, at least if they are not substantial enough.

It's of course possible to make additional commits later at review time, but many fixes most likely won't need any special mentioning and should just quickly make their way into the repository.

**Comment by oleg68** [comment]
@larspalo I agree with you
